src/cameraRoll.rei

```diff
@@ -0,0 +1,67 @@
+type node = {
+  .
+  "_type": string,
+  "group_name": string,
+  "image": {
+    .
+    "filename": string,
+    "height": int,
+    "isStored": bool,
+    "playableDuration": int,
+    "uri": string,
+    "width": int
+  },
+  "location": {
+    .
+    "altitude": float,
+    "heading": float,
+    "latitude": float,
+    "longitude": float,
+    "speed": float
+  },
+  "timestamp": float
+};
+
+type result = {
+  .
+  "edges": array({. "node": node}),
+  "page_info": {
+    .
+    "end_cursor": bool,
+    "has_next_page": bool,
+    "start_cursor": bool
+  }
+};
+
+type fileType =
+  | Video
+  | Photo;
+
+type groupType =
+  | All
+  | Album
+  | Event
+  | Faces
+  | Library
+  | PhotoStream
+  | SavedPhotos;
+
+type assetType =
+  | All
+  | Videos
+  | Photos;
+
+let saveToCameraRoll:
+  (~uri: string, ~type_: fileType=?, unit) => Js.Promise.t(string);
```

Comment:
I think it throws, so I suggest wrapping it in some result type so that the user is forced to handle it explicitly.

Reply:
hm, are you sure this is the correct way to handle Promise's errors?
I think the following code is correct (using `catch` in the Promise's chain):

```
CameraRoll.saveToCameraRoll(~uri="file:///sdcard/img.png", ())
|> Js.Promise.then_(uri => {
     Js.log(uri);
     Js.Promise.resolve();
   })
|> Js.Promise.catch(error => {
    /* do something with error */
    Js.Promise.resolve();
});
```

Reply:
Yes, but I think this is something we need to do inside the library, not exposing it to the user. So that you `catch` it and then resolve to `Js.Result.Error SomeErrorOrPossiblyUnit`. otherwise you resolve to `Js.Result.Ok something`. Thanks to that our promise is safe and users won't suffer from unhandled exceptions.

Reply:
@wokalski ok, I think I got it! please check this commit: https://github.com/reasonml-community/bs-react-native/pull/141/commits/e3b6825208ae83bb574cf926709e2bd8622542f2